src/codegen/compile.ml

```diff
@@ -11972,7 +12013,15 @@ and conclude_module env set_serialization_globals start_fi_o =
 
   let other_imports = E.get_other_imports env in
 
-  let memories = [nr {mtype = MemoryType {min = E.mem_size env; max = None}} ] in
+  let memories =
+    match E.mode env with
+    | Flags.ICMode
+    | Flags.RefMode ->
+       [nr {mtype = MemoryType {min = E.mem_size env; max = None}} ]
+    | Flags.WASIMode | Flags.WasmMode ->
+       [nr {mtype = MemoryType {min = E.mem_size env; max = None}};
+        nr {mtype = MemoryType {min = Int32.zero; max = None}} ] (* faux stable memory *)
```

Comment:
```suggestion
    nr {mtype = MemoryType {min = E.mem_size env; max = None}}
    ::
    match mode env with
    | Flags.WASIMode | Flags.WasmMode ->
      [ nr {mtype = MemoryType {min = Int32.zero; max = None}} ] (* faux stable memory *)
    | _ -> []
```

Reply:
Mini refactoring analogous to the other PR that builds upon this...

Reply:
Done by merge